devito/interfaces.py

```diff
@@ -0,0 +1,45 @@
+import numpy as np
+from sympy import IndexedBase
+
+
+class MatrixData(IndexedBase):
```

Comment:
I think `MatrixData` is misleading, since we are not actually providing matrix operations (eg. Mat-Vec multiplication). How about something like `GridData` or `DenseData` (as the counter-type to `SparseData`)?